pkg/manifests/manifests.go

```diff
@@ -78,17 +79,26 @@ func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.D
 
 	ds.Name = name
 
-	ds.Labels = map[string]string{
-		"app": "router",
-	}
+	ds.Spec.Template.Labels["router"] = name
 
 	ds.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: map[string]string{
-			"app": name,
+			"app":    "router",
```

Comment:
We could use `append` here, or else delete `spec.selector` from `assets/router/daemonset.yaml` if we are going to override `spec.selector.matchLabels` entirely and unconditionally.

Reply:
This also speaks to the external asset/code confusion. One argument I can imagine for managing the selector itself in code is that it contains dynamic values and also must match the labels to produce a valid DaemonSet. Maybe that could be incorporated into the manifest unit tests?

That said I guess I prefer removing `spec.selector` from the YAML.

Reply:
I went ahead and made both `spec.selector` and `spec.template.labels` managed entirely in code for now